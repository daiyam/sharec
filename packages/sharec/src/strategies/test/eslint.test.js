const path = require('path')
const { readFileSync } = require.requireActual('fs')
const { eslintStrategy } = require('strategies/eslint')

describe('strategy > eslint', () => {
  const eslint01 = require('fixtures/eslint/json/eslintrc_01.json')
  const eslint02 = require('fixtures/eslint/json/eslintrc_02.json')
  const eslint04 = require('fixtures/eslint/json/eslintrc_04.json')
  const eslint05 = require('fixtures/eslint/json/eslintrc_05.json')
  const eslintYaml01 = readFileSync(
    path.resolve(
      __dirname,
      '../../../test/fixtures/eslint/yaml/eslintrc_01.yml',
    ),
    'utf8',
  )
  const eslintYaml02 = readFileSync(
    path.resolve(
      __dirname,
      '../../../test/fixtures/eslint/yaml/eslintrc_02.yml',
    ),
    'utf8',
  )
  const eslintYaml04 = readFileSync(
    path.resolve(
      __dirname,
      '../../../test/fixtures/eslint/yaml/eslintrc_04.yml',
    ),
    'utf8',
  )
  const eslintYaml05 = readFileSync(
    path.resolve(
      __dirname,
      '../../../test/fixtures/eslint/yaml/eslintrc_05.yml',
    ),
    'utf8',
  )

  describe('json strategy', () => {
    it('should merge eslint json configs', () => {
      expect(eslintStrategy.mergeJSON(eslint01, eslint02)).toMatchSnapshot()
    })
  })

  describe('yaml strategy', () => {
    it('should merge eslint yaml configs', () => {
      expect(
        eslintStrategy.mergeYAML(eslintYaml01, eslintYaml02),
      ).toMatchSnapshot()
    })
  })

  describe('auto merge', () => {
    it('should automatically merge configs', () => {
      expect(
        eslintStrategy.merge('.eslintrc')(eslint01, eslint02),
      ).toMatchSnapshot()
      expect(
        eslintStrategy.merge('eslintrc.yml')(eslintYaml01, eslintYaml02),
      ).toMatchSnapshot()
    })
  })

  describe('uapplying JSON', () => {
    it('should remove applyed JSON config', () => {
      expect(eslintStrategy.unapplyJSON(eslint04, eslint05)).toMatchSnapshot()
    })
  })

  describe('uapplying YAML', () => {
    it('should remove applyed YAML config', () => {
      expect(
        eslintStrategy.unapplyYAML(eslintYaml04, eslintYaml05),
      ).toMatchSnapshot()
    })
  })

  describe('auto unapply', () => {
    it('should automatically unapply configs', () => {
      expect(
        eslintStrategy.unapply('.eslintrc')(eslint05, eslint04),
      ).toMatchSnapshot()
      expect(
        eslintStrategy.unapply('eslintrc.yml')(eslintYaml05, eslintYaml04),
      ).toMatchSnapshot()
    })
  })
})
