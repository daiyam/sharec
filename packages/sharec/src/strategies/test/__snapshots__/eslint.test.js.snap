// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`strategy > eslint auto merge should automatically merge configs 1`] = `
Object {
  "env": Object {
    "browser": true,
    "es6": true,
    "jest": true,
    "node": true,
  },
  "extends": "airbnb",
  "globals": Object {
    "bar": true,
    "foo": true,
  },
  "parserOptions": Object {
    "ecmaFeatures": Object {
      "spread": true,
    },
    "ecmaVersion": 7,
    "parser": "babel-eslint",
    "sourceType": "module",
  },
  "rules": Object {
    "max-len": Array [
      "warn",
      120,
      4,
      Object {
        "ignoreComments": true,
        "ignoreStrings": false,
        "ignoreUrls": false,
      },
    ],
    "no-console": 0,
    "space-before-function-paren": 1,
  },
}
`;

exports[`strategy > eslint auto merge should automatically merge configs 2`] = `
"env:
  browser: true
  es6: true
  node: true
  jest: true
globals:
  foo: true
  bar: true
extends: airbnb
parserOptions:
  parser: babel-eslint
  ecmaVersion: 7
  ecmaFeatures:
    spread: true
  sourceType: module
rules:
  space-before-function-paren: 1
  no-console: 0
  max-len:
    - warn
    - 120
    - 4
    - ignoreUrls: false
      ignoreComments: true
      ignoreStrings: false
"
`;

exports[`strategy > eslint auto unapply should automatically unapply configs 1`] = `
Object {
  "env": Object {
    "node": false,
  },
  "extends": "standard",
  "globals": Object {
    "baz": true,
  },
  "rules": Object {
    "max-len": Array [
      "warn",
      80,
      2,
    ],
    "sort-imports": 0,
  },
}
`;

exports[`strategy > eslint auto unapply should automatically unapply configs 2`] = `
"env:
  node: false
extends: standard
globals:
  baz: true
rules:
  max-len:
    - warn
    - 80
    - 2
  sort-imports: 0
"
`;

exports[`strategy > eslint json strategy should merge eslint json configs 1`] = `
Object {
  "env": Object {
    "browser": true,
    "es6": true,
    "jest": true,
    "node": true,
  },
  "extends": "airbnb",
  "globals": Object {
    "bar": true,
    "foo": true,
  },
  "parserOptions": Object {
    "ecmaFeatures": Object {
      "spread": true,
    },
    "ecmaVersion": 7,
    "parser": "babel-eslint",
    "sourceType": "module",
  },
  "rules": Object {
    "max-len": Array [
      "warn",
      120,
      4,
      Object {
        "ignoreComments": true,
        "ignoreStrings": false,
        "ignoreUrls": false,
      },
    ],
    "no-console": 0,
    "space-before-function-paren": 1,
  },
}
`;

exports[`strategy > eslint uapplying JSON should remove applyed JSON config 1`] = `
Object {
  "env": Object {
    "node": false,
  },
  "extends": "standard",
  "globals": Object {
    "baz": true,
  },
  "rules": Object {
    "max-len": Array [
      "warn",
      80,
      2,
    ],
    "sort-imports": 0,
  },
}
`;

exports[`strategy > eslint uapplying YAML should fully unapply YAML config and return empty object 1`] = `
"globals:
  baz: true
env:
  node: false
extends: standard
rules:
  max-len:
    - warn
    - 80
    - 2
  sort-imports: 0
"
`;

exports[`strategy > eslint uapplying YAML should remove applyed YAML config 1`] = `
"env:
  node: false
extends: standard
globals:
  baz: true
rules:
  max-len:
    - warn
    - 80
    - 2
  sort-imports: 0
"
`;

exports[`strategy > eslint yaml strategy should merge eslint yaml configs 1`] = `
"env:
  browser: true
  es6: true
  node: true
  jest: true
globals:
  foo: true
  bar: true
extends: airbnb
parserOptions:
  parser: babel-eslint
  ecmaVersion: 7
  ecmaFeatures:
    spread: true
  sourceType: module
rules:
  space-before-function-paren: 1
  no-console: 0
  max-len:
    - warn
    - 120
    - 4
    - ignoreUrls: false
      ignoreComments: true
      ignoreStrings: false
"
`;
